pkg/controller/system/controller.go

```diff
@@ -49,22 +49,39 @@ import (
 	styraclient "github.com/crossplane-contrib/provider-styra/pkg/client"
 )
 
-const (
-	helmValuesAssetType            = "helm-values"
-	helmValuesConnectionDetailsKey = "helmValues"
+type connector struct {
+	kube        client.Client
+	newClientFn func(transport runtime.ClientTransport, formats strfmt.Registry) *styra.StyraAPI
+}
+
+type external struct {
+	client *styra.StyraAPI
+	kube   client.Client
+}
+
+type systemDetails struct {
+	AssetType                  string
```

Comment:
If you want to make abstractions you could also add funcs to `SystemParameters` in `apis`, such as `HasLabels`, `ConnectionDetailsKey` and `AssetType`.

In case of `errorGetValuesMessage` I would replace it with a generic `cannot get asset %s` in the controller.